datamodel/odata-client/src/main/java/com/sap/cloud/sdk/datamodel/odata/client/request/ODataRequestResultFactory.java

```
@@ -0,0 +1,42 @@
+package com.sap.cloud.sdk.datamodel.odata.client.request;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.entity.BufferedHttpEntity;
+import org.apache.http.message.BasicHttpResponse;
+
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+
+/**
+ * Enum representing the strategy for buffering HTTP responses.
+ */
+@FunctionalInterface
+interface ODataRequestResultFactory
+{
+    /**
+     * Strategy that does not buffer the response.
+     */
+    ODataRequestResultFactory WITHOUT_BUFFER = ODataRequestResultResource::new;
+
+    /**
+     * Strategy that buffers the response by creating a copy of it.
+     */
+    ODataRequestResultFactory WITH_BUFFER = ( oDataRequest, httpResponse, httpClient ) -> {
+        final BasicHttpResponse copy = new BasicHttpResponse(httpResponse.getStatusLine());
+        Option.of(httpResponse.getLocale()).peek(copy::setLocale);
+        Option.of(httpResponse.getAllHeaders()).peek(copy::setHeaders);
+        Option
+            .of(httpResponse.getEntity())
+            .peek(entity -> Try.run(() -> copy.setEntity(new BufferedHttpEntity(entity))));
```

This currently drops an important warning log, please add it back in. It's highly likely that any IOException here will cause a failure later, and without the log the exception may be lost.

Unfortunately, logging is always annoying in interfaces, so maybe better to move this to class, e.g. with an inner interface?

---

Thanks for pointing it out!
I missed this regression.